ambry-commons/src/test/java/com.github.ambry.commons/BlobIdTest.java

```diff
@@ -13,119 +13,440 @@
  */
 package com.github.ambry.commons;
 
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.clustermap.ClusterMap;
 import com.github.ambry.clustermap.MockClusterMap;
 import com.github.ambry.clustermap.MockPartitionId;
 import com.github.ambry.clustermap.PartitionId;
 import com.github.ambry.utils.ByteBufferInputStream;
 import java.io.DataInputStream;
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Random;
 import java.util.UUID;
 import org.apache.commons.codec.binary.Base64;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+import static com.github.ambry.clustermap.ClusterMapUtils.*;
+import static com.github.ambry.commons.BlobId.*;
+import static com.github.ambry.utils.Utils.*;
 import static org.junit.Assert.*;
 
 
+/**
+ * Unit tests for {@link BlobId}. Using {@link BlobIdV2} builds a {@link BlobId}
+ * in v2 equivalent to setting {@link BlobId#CURRENT_VERSION} to be {@link BlobId#BLOB_ID_V2}.
+ */
+@RunWith(Parameterized.class)
 public class BlobIdTest {
+  private static final Random random = new Random();
+  private final short version;
+  private byte referenceFlag;
+  private byte referenceDatacenterId;
+  private short referenceAccountId;
+  private short referenceContainerId;
+  private ClusterMap referenceClusterMap;
+  private PartitionId referencePartitionId;
+
+  /**
+   * Running for both {@link BlobId#BLOB_ID_V1} and {@link BlobId#BLOB_ID_V2}
+   * @return an array with both {@link BlobId#BLOB_ID_V1} and {@link BlobId#BLOB_ID_V2}
+   */
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    return Arrays.asList(new Object[][]{{BLOB_ID_V1}, {BLOB_ID_V2}});
+  }
+
+  /**
+   * Constructor with parameter to be set.
+   * @param version The version for BlobId to test.
+   */
+  public BlobIdTest(short version) {
+    this.version = version;
+  }
+
+  /**
+   * Initialization before each unit test.
+   * @throws Exception Any unexpected exception.
+   */
+  @Before
+  public void init() throws Exception {
+    byte[] bytes = new byte[2];
```

Comment:
why not set all of these in the constructor? Maybe you can then make most class variables `final`